test/parallel/test-stream-readable-async-iterators.js

```diff
@@ -484,6 +484,22 @@ async function tests() {
       assert.strictEqual(e, err);
     })()]);
   }
+
+  {
+    const r = new Readable({
+      read() {
+      },
+      destroy(err, callback) {
+        setTimeout(() => callback(new Error('asd')), 1);
+      }
+    });
+
+    r.destroy();
+    const it = r[Symbol.asyncIterator]();
+    it.next().catch(common.mustCall((err) => {
+      assert.strictEqual(err.message, 'asd');
```

Comment:
Nit: instead of checking the error message, what about checking for reference equality?

Reply:
Yea, probably better... but I don't think it matters.

Reply:
It would catch issues in case it is refactored at a later point?

Reply:
Sure